graph/tarjan.cpp

```diff
@@ -0,0 +1,227 @@
+#include<iostream> /// for general input and output
```

Comment:
Please describe how the algorithm works, a Wikipedia or any other source for algorithm explanation, what the algorithm does, etc..